.github/workflows/reusable_basic.yml

```diff
@@ -112,23 +113,17 @@ jobs:
             disable_hwloc: 'OFF'
             link_hwloc_statically: 'ON'
     runs-on: ${{matrix.os}}
-
+    container:
+      image: ${{ matrix.os == 'ubuntu-20.04' && 'ghcr.io/rbanka1/umf2-ubuntu-20.04:latest' || matrix.os == 'ubuntu-22.04' && 'ghcr.io/rbanka1/umf2-ubuntu-22.04:latest' || matrix.os == 'ubuntu-24.04' &&  'ghcr.io/rbanka1/umf2-ubuntu-24.04:latest'}}
+      options: --user root --privileged  
+      volumes:
+        - /home/runner/work/unified-memory-framework/unified-memory-framework:/home/runner/work/unified-memory-framework/unified-memory-framework
     steps:
     - name: Checkout
       uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       with:
         fetch-depth: 0
 
-    - name: Install apt packages
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y clang cmake libnuma-dev lcov
-
-    - name: Install TBB apt package
```

Comment:
this step here gives us chance to test our project without TBB. Perhaps it should be re-design now, so we remove the TBB from the OS if `matrix.install_tbb == 'OFF'`